dimod/constrained/constrained.py

```diff
@@ -1783,7 +1789,8 @@ def to_file(self, *, spool_size: int = int(1e9)) -> tempfile.SpooledTemporaryFil
         write_header(file, CQM_MAGIC_PREFIX, data, version=(1, 3))
 
         # write the values
-        with zipfile.ZipFile(file, mode='a') as zf:
+        kwargs = dict(compression=zipfile.ZIP_DEFLATED) if compress else dict()
+        with zipfile.ZipFile(file, mode='a', **kwargs) as zf:
```

Comment:
My only doubt in this PR was about the compression method and compression level to use.

We have `bzip2` and `lzma` at our disposal, in addition to the default `gzip`.

[Looks like](https://tukaani.org/lzma/benchmarks.html) `bzip2` provides 15% better compression over `gzip`, but it's ~8x slower at (de)compression.

`lzma` halves the archive size over `gzip`, but it's a lot slower than `bzip2` during compression, although roughly as fast as `gzip` during decompression.

`gzip`/deflate dominates in memory overhead.

In addition, `bzip2`/`lzma` modules might not be available on all systems, especially if user self-compiled Python, because system packages need to installed beforehand.

Taking all that into consideration, looks like `ZIP_DEFLATED` with the default compression level (6 - balance between speed and size) is the optimal choice here. It's widely supported, it's fast, with minimal memory overhead, and provides decent compression (at least to handle obviously redundant sequences).